_posts/2019-02-11-configurable-builds-part-1.md

```diff
@@ -0,0 +1,284 @@
+---
+layout: posts
+title: Configurable Builds - Part 1
+authors:
+ - gregestren
+---
+
+One of Bazel's long-term goals is to make `$ bazel build //:all` "*just work*"
+so that every target builds "the right way", for whatever platform(s) you care
+about, using only flags that are genuinely interesting to you. This two-part
+series discusses the challenges involved and steps we're taking to get there.
+
+This is a deeper dive into themes covered in Bazel's [configurability roadmap]
+(https://bazel.build/roadmaps/configuration.html).
+
+## Motivation
+
+The larger your project gets, the more likely you are to have to build it
+different ways.
+
+Maybe you need Android, iOS, and desktop versions of your app. Maybe you build
+C++ for different platforms. Or maybe you maintain a popular JavaScript library
+that users only want parts of to keep their .js files small.
+
+These are examples of *configuration*: the process of building the same *code*
+with different *settings* to customize it for specific needs. In Bazel-speak,
+this means building a
+[target](https://docs.bazel.build/versions/master/build-ref.html#targets) with
+a set of
+[flags](https://docs.bazel.build/versions/master/user-manual.html#flags-options):
+
+```sh
+$ bazel build //my:cc_binary --cpu=x86
+$ bazel build //my:cc_binary --cpu=arm
+$ bazel build //my:android_binary --android_sdk=@androidsdk//:sdk-28
+```
+
+Bazel's *configurability work* is an ongoing effort to make these tasks simple,
+flexible, and powerful. 
+
+## The Problem
+
+Bazel is a
+[powerful](https://docs.bazel.build/versions/master/bazel-vision.html) build
+tool that's especially suited for large codebases with
+[multiple languages](https://blog.bazel.build/2018/12/05/multilanguage-build-system.html).
+But it grew out of Google, which historically wrote code for fleets of identical
```

Comment:
I'm still not a huge fan of this phrasing, on two counts. I think "grew out of" sounds a little strange relative to the tone of the rest of the article, and it also says that Google itself writes code. What about "But it was originally created for Google engineers, who wrote code for..."

Reply:
Okay, varied it up. And I agree. "grow" is a weird word  - I ultimately just wanted to express evolution, which I think this new wording does equally well.